### PR TITLE
Some code cleanup and a pings option.

### PR DESCRIPTION
**Pull request by andrewdavidmackenzie**
Use of constants. Print out more info using the actual value set not hardcoded in strings.
Added option to specify number of pings to perform.
Print out the version number of the program at start.
Store results in variables ready to print/send a summary.
Removed commented code.
Moved some of the string output relevant to Server into server class.
Added a TOTO.md file to keep track of ideas and things to improve.
Bumped the version number for my fork.

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -11,3 +11,5 @@ license = "MIT"
 time = "0.1"
 byteorder = "0.5"
 getopts = "0.2"
+log = "*"
+log4rs = { version = "*", features = ["yaml"] }
```
Comment (Kilobyte22) [inline Cargo.toml:15]: I'd appreciate if you actually specified a version.

File: src/main.rs [hunk 2]
```diff
@@ -73,39 +96,35 @@ fn print_usage(program: &str, opts: Options) {
...
-fn launch_server(port: u16, listen: &str) {
-    println!("Listening...");
-    let s = server::TestServer::new(port, listen);
-    s.listen();
-}
-
 fn format_speed(speed: f64) -> String {
     let mut speed = speed;
-    let units = ["bit/s", "kbit/s", "Mbit/s", "Gbit/s", "Tbit/s?!"];
+    let units = ["bit/s", "Kbit/s", "Mbit/s", "Gbit/s", "Tbit/s"];
```
Comment (Kilobyte22) [inline src/main.rs:122]: I actually checked and it's kbit. Also refer to https://en.wikipedia.org/wiki/Data_rate_units

**Comment by Kilobyte22** [comment]
Looked over it, looks pretty good. Just about to go to bed, so haven't actually tested it (hence no merge). I have however some minor things I'd prefer differently (see inline comments). Other than that definitely good changes.

**Comment by Kilobyte22** [comment]
I am just curious. What is the motivation behind using a full logging system here? This program is not the kind of program which logging systems are made fore (namely: daemons). IMO thats way overkill.

**Comment by andrewdavidmackenzie** [comment]
I'dike to run one copy on a server (probably started at init, demonized)
and another copy on clients/devices that do the test regularly and send
results to a data store.

Logging makes sense for both to figure out what happened and failures later.

My goals here maybe diverging from what you want to do, which is OK ....

I can rename, and avoid future PRs if you prefer.

Andrew

File: src/main.rs (hunk 2 above)
Comment (andrewdavidmackenzie) [inline src/main.rs:122]: Sorry about that! Got confused. You are correct.

**Comment by andrewdavidmackenzie** [comment]
To be honest, I didn't mean to include all the subsequent commits in the PR - but I forgot to change to a feature branch after creating the original PR - so I can see why your overwhelmed!

I'll finish the change I'm working on now, and then switch to feature branches (hence separate PRs) for future changes.

**Pull request closed without merging**